### PR TITLE
Common: Remove async logging

### PR DESCRIPTION
**Pull request by yuriks**
It provided a large increase in complexity of the logging system while
having a negligible performance impact: the usage patterns of the ring
buffer meant that each log contended with the logging thread, causing
it to effectively act as a synchronous extra buffering.

Also removed some broken code related to filtering of subclasses which
was broken since it was introduced. (Which means no one ever used that
feature anyway, since, 8 months later, no one ever complained.)

### CODE REVIEW

**Comment by yuriks** [comment]
Updated with the assert change.

File: src/common/assert.h [hunk 1]
```diff
@@ -28,19 +29,14 @@ static void assert_noinline_call(const Fn& fn) {
     exit(1); // Keeps GCC's mouth shut about this actually returning
 }
 
-// TODO (yuriks) allow synchronous logging so we don't need printf
 #define ASSERT(_a_) \
     do if (!(_a_)) { assert_noinline_call([] { \
-        fprintf(stderr, "Assertion Failed!\n\n  Line: %d\n  File: %s\n  Time: %s\n", \
-                     __LINE__, __FILE__, __TIME__); \
+        LOG_CRITICAL(Debug, "Assertion Failed!"); \
```
Comment (bunnei) [inline src/common/assert.h]: Why are you getting rid of line/file/..? I find that stuff useful.

Comment (yuriks) [inline src/common/assert.h]: All that info is already included in the message by the log system.

**Comment by bunnei** [comment]
Woo lots of deletions!
Looks fine, except my one comment. Fwiw, is it worth checking if the performance impact is still negligible even without the SW renderer bottleneck?

File: src/common/assert.h (hunk 1 above)
Comment (bunnei) [inline src/common/assert.h]: Ah right...

**Comment by linkmauve** [comment]
You should also remove the now-unused concurrent_ring_buffer.h

**Comment by yuriks** [comment]
Updated

**Comment by archshift** [comment]
LGTM if CI passes.

**Comment by yuriks** [comment]
360 noscope merge

**Pull request merged**